lib/customer.rb

```diff
@@ -0,0 +1,34 @@
+class Customer
+
+  attr_reader :basket
+
+  def initialize
+    @basket = []
+  end
+
+  def view_menu(takeaway)
+    takeaway.menu
+  end
+
+  def add_item(takeaway, _item)
+    fail 'Sorry that item is not on the menu' if takeaway.menu.key?(_item) == false
+    @basket << _item
+  end
+
+  def place_order(takeaway)
+    puts "Send us your number to confirm the order"
+    @customer_number = "+447850122007"
+    takeaway.receive_order
+  end
+
+  def bill(takeaway)
+    final_bill = []
+      self.basket.each do |item|
+        if takeaway.menu.has_key?(item)
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`.<br>Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.